### PR TITLE
REALMC-6874: Add native String padStart, trimStart, & trimEnd methods

### PR DESCRIPTION
**Pull request by makesitgo**

### CODE REVIEW

File: builtin_string.go [hunk 1]
```diff
@@ -448,6 +448,44 @@ func builtinString_toUpperCase(call FunctionCall) Value {
...
+
+	var builder strings.Builder
+	for delta > 0 {
+		prefix := padString
+		if int64(len(padString)) > delta {
+			prefix = padString[:delta]
+		}
+
+		if _, err := builder.WriteString(prefix); err != nil {
+			// error handling?
```
Comment (makesitgo) [inline builtin_string.go:479]: ⚠️ ... not sure what to do here

File: inline.go [hunk 2]
```diff
@@ -1966,6 +1966,29 @@ func _newContext(runtime *_runtime) {
...
+			class:       "Function",
+			objectClass: _classObject,
+			prototype:   runtime.global.FunctionPrototype,
+			extensible:  true,
+			property: map[string]_property{
+				"length": _property{
+					mode: 0,
+					value: Value{
+						kind:  valueNumber,
+						value: 2, // what is this value?
```
Comment (makesitgo) [inline inline.go:1980]: ⚠️ ... also blindly went with `2` here, not sure if that's correct

File: builtin_string.go (hunk 1 above)
Comment (makesitgo) [reply builtin_string.go:479]: Maybe something along the lines of:

```go
panic(call.runtime.panicTypeError())
```

File: inline.go (hunk 2 above)
Comment (tsedgwick) [reply inline.go:1980]: this is the number of arguments

File: builtin_string.go (hunk 1 above)
Comment (tsedgwick) [reply builtin_string.go:479]: yeah this should panic to allow the error to be thrown

File: builtin_string.go [hunk 3]
```diff
@@ -448,6 +448,44 @@ func builtinString_toUpperCase(call FunctionCall) Value {
...
+
+	if arg0, ok := call.getArgument(0); ok {
+		targetLength = arg0.number().int64
+	}
+
+	if arg1, ok := call.getArgument(1); ok {
+		padString = arg1.string()
+	}
+
+	delta := targetLength - int64(len(target))
```
Comment (tsedgwick) [inline builtin_string.go:466]: [nit] we can move this right after the arg0 check before we check the second arg:

```
if arg0, ok := call.getArgument(0); ok {
		targetLength = arg0.number().int64
	}
delta := targetLength - int64(len(target))
if delta <= 0 {
		return toValue_string(target)
	}


if arg1, ok := call.getArgument(1); ok {
		padString = arg1.string()
	}

File: builtin_string.go [hunk 4]
```diff
@@ -448,6 +448,44 @@ func builtinString_toUpperCase(call FunctionCall) Value {
...
+	if delta <= 0 {
+		return toValue_string(target)
+	}
+
+	var builder strings.Builder
+	for delta > 0 {
+		prefix := padString
+		if int64(len(padString)) > delta {
```
Comment (tsedgwick) [inline builtin_string.go:474]: let's make padStringLen a var outside of the loop so we don't have to continuously cast it to an int64.

**Review by tsedgwick — CHANGES_REQUESTED** [review]
just a few nits

**Review by tsedgwick — APPROVED** [review]
🆒

**Pull request merged**